src/pep8ext_naming.py

```diff
@@ -437,6 +437,5 @@ def _extract_names(assignment_target):
 
 def is_mixed_case(name):
     stripped_name = name.strip('_')
-    head = stripped_name[:1]
-    tail = stripped_name[1:]
-    return head.islower() and tail and not (tail.islower() or tail.isdigit())
+    head_is_lower = stripped_name[:1].islower()
+    return head_is_lower and any(map(str.isupper, stripped_name[1:]))
```

Comment:
map is going to be slower here than a genexpr or list comprehension. If we're calling this function a lot, it might make sense to optimize for that.

Reply:
I ran a simple test on python 3.6 and map seemed to be slightly faster than generator expression, but I guess that it would be the other way around on 2.7.

my test: https://gist.github.com/5j9/c75d984cede3c0280788a6f27b69d9f9

As far as the performance goes, `tail.lower() != tail` was the best that I could come up with.

Reply:
`str.isupper()` is a builtin (defined in C) function. `map()` tends to perform best in that case.